### PR TITLE
Validations for the WaitForChange arguments

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json [hunk 1]
```diff
@@ -25,6 +24,7 @@
           "type": "platform"
         }
       }
-    }
+    },
+    "net451": {}
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json]: Any reason for this change?

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs [hunk 2]
```diff
@@ -362,6 +362,12 @@ public virtual Task<bool> IsServerAvailableAsync(CancellationToken cancellationT
         /// Alternatively, use <seealso cref="ProjectMetadataChanged"/> or <seealso cref="InstanceMetadataChanged"/>, respectively.
         /// </para>
         /// </remarks>
+        /// <exception cref="ArgumentException">
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs]: I'd document both of these in a single element.

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json (hunk 1 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json]: Yes, there appears to be some bug with VS where it won't load the tests into the test explorer if they are in the other order.

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs (hunk 2 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs]: Done (below as well)

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json]: Okay, that's bizarre - but may explain the issues I've been having. (I've been just running from the command line for a while.)

If the rest of the change were larger, I'd suggest splitting the commit, but I don't think it's worth it.

**Comment by jskeet** [comment]
LGTM.

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json (hunk 1 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/project.json]: Yeah, good point. In the future I'll try to separate out something like this. For context, here is where I found the workaround: https://github.com/xunit/xunit/issues/900

**Pull request merged**